src/core/core.element.ts

```diff
@@ -28,16 +28,18 @@ export default class Element<T extends AnyObject = AnyObject, O extends AnyObjec
    * @param props - properties to get
    * @param [final] - get the final value (animation target)
    */
-  getProps<P extends (keyof T)[]>(props: P, final?: boolean): Pick<T, P[number]> {
+  getProps<P extends string>(props: P[], final?: boolean): Partial<Record<P, unknown>>;
+  getProps<P extends (keyof T)[]>(props: P, final?: boolean): Pick<T, P[number]>;
```

Comment:
Should this be `P[number]` or `P[T]`?

Reply:
@etimberg  Would you please be more accurate? I don't understand what you mean.

Reply:
I'm not sure what type `P[number]` is. If the keys are strings, then `P[number]` seems statically undefined. Usually when I've written code like this before in TS, it would be something like `Pick<T, P[T]>`. I might be totally misunderstanding this code though

Reply:
@etimberg Example:
```ts
getProps(['x', 'y'])
P[number] === ('x' | 'y')
```

Reply:
gotcha, makes sense 👍